cmd/create/cluster/cmd.go

```diff
@@ -2011,26 +2018,47 @@ func run(cmd *cobra.Command, _ []string) {
 			os.Exit(1)
 		}
 		excludedSubnetsDueToCidr := []string{}
+		excludedSubnetsDueToPublic := []string{}
 		for _, subnet := range initialSubnets {
+			skip := false
 			subnetIP, subnetNetwork, err := net.ParseCIDR(*subnet.CidrBlock)
 			if err != nil {
 				r.Reporter.Errorf("Unable to parse subnet CIDR")
 				os.Exit(1)
 			}
 
-			if machineNetwork.Contains(subnetIP) &&
+			if !(machineNetwork.Contains(subnetIP) &&
```

Comment:
Should it be `||` here?
```suggestion
			if !(machineNetwork.Contains(subnetIP) ||
```
Maybe we can simplify the condition to improve readability.

Reply:
Maybe we can create a new function `isValidCidr` and then encapsulate the previous conditions we had in the function?